### ISSUE TITLE
Incorrect 'commands' generated for a ConsoleApplication

### ISSUE THREAD

**Issue by egamma** [OmniSharp/generator-aspnet#335]
version: generator-aspnet@0.0.73

Generate a ConsoleApplication.

This generates an incorrect 'commands':
`
  "commands": {
    "ConsoleApplication": "ConsoleApplication"
  },
`
It should be:
`
  "commands": {
    "ConsoleApplication": "run"
  },
`

**Comment by sayedihashimi** [comment]
Thanks @egamma, by default we do create the command in the form ``` "commands": { "ConsoleApplication": "ConsoleApplication" }``` which is consistent with what Visual Studio does.

I tried both ``` "commands": { "ConsoleApplication": "ConsoleApplication" }, ``` and ```"commands": { "ConsoleApplication": "run" }``` and they both run successfully on both windows/mac and look to do the same thing.

I discussed with @glennc and this change was by design. Are you have any issues with the current implementation?

**Comment by peterblazejewicz** [comment]
@sayedihashimi @glennc
https://github.com/aspnet/Tooling/issues/161

**Comment by egamma** [comment]
On beta7 the launch fails, when changing the command to 'run' then the launch succeeds.

`
System.InvalidOperationException: Unable to load application or execute command 'ConsoleApplication'. Available commands: ConsoleApplication.
   at Microsoft.Dnx.ApplicationHost.Program.ThrowEntryPointNotfoundException(DefaultHost host, String applicationName, Exception innerException)
   at Microsoft.Dnx.ApplicationHost.Program.ExecuteMain(DefaultHost host, String applicationName, String[] args)
   at Microsoft.Dnx.ApplicationHost.Program.Main(String[] args)
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at Microsoft.Dnx.Runtime.Common.EntryPointExecutor.Execute(Assembly assembly, String[] args, IServiceProvider serviceProvider)
   at Microsoft.Dnx.Host.Bootstrapper.RunAsync(List`1 args, IRuntimeEnvironment env, FrameworkName targetFramework)
   at Microsoft.Dnx.Host.RuntimeBootstrapper.ExecuteAsync(String[] args, FrameworkName targetFramework)
   at Microsoft.Dnx.Host.RuntimeBootstrapper.Execute(String[] args, FrameworkName targetFramework)
`

**Comment by peterblazejewicz** [comment]
@egamma Thanks, we will check.

@sayedihashimi 
I can only get that error reported when the command name is e.g. misspelled, e.g. `dnx ConsoleApppplication`:
```
System.InvalidOperationException: Unable to load application or execute command 'ConsoleApppplication'. Available commands: ConsoleApplication.
```
It looks that dnx does not find correct command on Erich machine.

**Comment by sayedihashimi** [comment]
@egamma it's working for me on both mac and windows. See below for mac.
```
Sayeds-MacBook-Pro:yotest3 sayedhashimi$ cd ConsoleApplication/
Sayeds-MacBook-Pro:ConsoleApplication sayedhashimi$ dnx ConsoleApplication
Hello World
```

I think @peterblazejewicz is on to something here. Are you sure that you typed the command name exactly as in `project.json`? The error you reported is surfaced when the command name is not found. If so then did you upgrade `dnvm` along with dnx? If not maybe that could be the cause.

This does bring us back to the issue that @peterblazejewicz brought up https://github.com/aspnet/Tooling/issues/161. Perhaps keeping it as `dnx run` would be better than using the project name as the command name? @egamma if you could let us know your thoughts on this at https://github.com/aspnet/Tooling/issues/161 I'd appreciate it.

**Comment by sayedihashimi** [comment]
Hmm I just looked at the error message provided more closely 

>System.InvalidOperationException: Unable to load application or execute command 'ConsoleApplication'. Available commands: ConsoleApplication.

So it looks like you have the command name exactly as it's shown in Available commands: I'm not sure what may cause this.

@davidfowl any idea what might be happening here?

**Comment by egamma** [comment]
regarding the dnx command. This is what I get from where dnx:
`
C:\Users\egamma\.dnx\runtimes\dnx-clr-win-x86.1.0.0-beta7\bin\dnx.exe
`

Verified that I didn't mistype by copying and pasting the name of the command from the project.json.

**Comment by sayedihashimi** [comment]
ok let's see if @davidfowl has any ideas.

**Comment by egamma** [comment]
Just making sure that I provide all the data. Regarding dnvm update - these were my installation steps:
deleted the .dnx folder
installed dnvm
set DNX_UNSTABLE_FEED=”https://www.myget.org/F/aspnetliterelease/api/v2”
dnvm upgrade

**Comment by davidfowl** [comment]
Why are you using any feed? beta7 is on nuget now.

**Comment by egamma** [comment]
Well, how do I know (all I want is to test whether VS Code supports beta7)?

I tried again with the official beta7. I still see the same problem. This is the dnvm upgrade output

```
C:\Users\egamma>dnvm upgrade
Determining latest version
Downloading dnx-clr-win-x86.1.0.0-beta7 from https://www.nuget.org/api/v2
Installing to C:\Users\egamma\.dnx\runtimes\dnx-clr-win-x86.1.0.0-beta7
Adding C:\Users\egamma\.dnx\runtimes\dnx-clr-win-x86.1.0.0-beta7\bin to process PATH
Adding C:\Users\egamma\.dnx\runtimes\dnx-clr-win-x86.1.0.0-beta7\bin to user PATH
Native image generation (ngen) is skipped. Include -Ngen switch to turn on native image generation to improve application startup time.
Setting alias 'default' to 'dnx-clr-win-x86.1.0.0-beta7'
```

**Comment by peterblazejewicz** [comment]
a note: review this during  beta8

**Comment by peterblazejewicz** [comment]
just as update: the `beta8` commands in `project.json` are the same as in `beta7`:
https://github.com/OmniSharp/generator-aspnet/blob/beta8/templates/projects/console/project.json#L13

**Comment by smithrobs** [comment]
I just hit this with a clean OS X dnx install. Same error as above:

    Error: Unable to load application or execute command 'ConsoleApplication'. Available commands: ConsoleApplication.

Switching to

    "commands": { "ConsoleApplication": "run" }, 

works. `which dnx` returns

    /Users/<user>/.dnx/runtimes/dnx-coreclr-darwin-x64.1.0.0-beta8/bin/dnx

**Comment by sayedihashimi** [comment]
Is this still an issue?

**Comment by smithrobs** [comment]
Still happening.

I upgraded to 1.0.0-rc1-16202... `which dnx` returns

    Users/<user>/.dnx/runtimes/dnx-mono.1.0.0-rc1-final/bin/dnx

**Comment by smithrobs** [comment]
I believe I've stumbled into an alternate solution, but I don't know if it's a bug or by design.

The console app I had generated resides inside a directory named `TestApp`. The project.json command section is as follows:

    "commands": { "ConsoleApplication": "ConsoleApplication" }, 

That will generate the dnx "Unable to load application or execute command" error.

We already know that if we change it to

    "commands": { "ConsoleApplication": "run" }, 

it works, but this is not the way it should be generated.

Now, if I change the command to match the **directory name** like so

    "commands": { "ConsoleApplication": "TestApp" }, 

it runs the app. So it seems like the command to run is tied to the name of the parent directory? This may be more of a `yo aspnet` generator issue?
